source/AI.cpp

```diff
@@ -1715,9 +1715,11 @@ void AI::PickUp(Ship &ship, Command &command, const Body &target)
 
 void AI::DoSurveillance(Ship &ship, Command &command) const
 {
-	const shared_ptr<Ship> &target = ship.GetTargetShip();
-	if(target && (!target->IsTargetable() || target->GetSystem() != ship.GetSystem()))
-		ship.SetTargetShip(shared_ptr<Ship>());
+	// Since DoSurveillance is called after target-seeking and firing, if this
+	// ship has a target, that target is guaranteed to be targetable.
+	shared_ptr<Ship> target = ship.GetTargetShip();
+	if(target && (target->GetSystem() != ship.GetSystem() || target->IsEnteringHyperspace()))
+		target.reset();
```

Comment:
You should probably also call `ship.SetTargetShip(shared_ptr<Ship>())` here so the ship clears the bad target instead of holding on to it frame after frame.

Reply:
Addressed.